### DISCUSSION TITLE
How can I change azure AI Search result reference ?

### DISCUSSION THREAD

**Discussion by Cobra86** [microsoft/semantic-kernel#7406] [Q&A] [answered]
Is it possible to change the AI search reference from [doc2] to something like filename or the document id ?
I tried to set the prompt in the RoleInformation property but i didn't get same output and sometime the AI will only return this reference [doc3].

**Comment by dmytrostruk** [comment]
@Cobra86 Azure .NET SDK returns citation information associated with chat response:
![image](https://github.com/user-attachments/assets/3d4c8518-5f33-4fef-91ce-89f5082c8a81)

At the moment, we don't expose citation information in Semantic Kernel, but we can add it in response metadata if needed. Please let me know if this would be helpful and I will create GitHub Issue for that. 

But even if we will expose citation info in response metadata, there is still no relationship between `[doc1]` mark and specific citation returned. I assume you will have to rely on citation index in response array. In all cases, you will have to do the replacement of `[doc1]` to the desired info on your side, since this response is coming from Azure service, and we don't have a control over it.

**Reply by Cobra86** [reply]
Yes, please. This is very useful. My project is similar to the demo available at https://github.com/Azure-Samples/azure-search-openai-demo-csharp. 
The majority of the projects I'm working on involve retrieving documents from SharePoint or Azure AI Search and summarising them. Especially in finance, they want to see the references to sources to double check the summary or response.

**Reply by dmytrostruk** [reply, marked as answer]
Sounds good, I created an issue for this: https://github.com/microsoft/semantic-kernel/issues/7438
Meanwhile, I will close this discussion, so GitHub Issue will be the only source of truth for the progress of this request. Thanks!
